src/Controller/UserController.php

```diff
@@ -4,64 +4,88 @@
 
 use App\Entity\User;
 use App\Form\UserType;
+use Doctrine\Persistence\ManagerRegistry;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
 use Symfony\Component\Routing\Annotation\Route;
 
 class UserController extends AbstractController
 {
-    /**
-     * @Route("/users", name="user_list")
-     */
-    public function listAction()
+
+    #[Route('/users', name: 'app_user_list')]
+    public function listAction(ManagerRegistry $doctrine): Response
     {
-        return $this->render('user/list.html.twig', ['users' => $this->getDoctrine()->getRepository('AppBundle:User')->findAll()]);
+        return $this->render('user/list.html.twig', ['users' => $doctrine->getRepository(User::class)->findAll()]);
     }
 
     #[Route('/users/create', name: 'app_user_create')]
-    public function createAction(Request $request)
+    public function createAction(ManagerRegistry $doctrine, UserPasswordHasherInterface $userPasswordHasher, Request $request): RedirectResponse|Response
     {
         $user = new User();
         $form = $this->createForm(UserType::class, $user);
 
         $form->handleRequest($request);
 
-        if ($form->isValid()) {
-            $em = $this->getDoctrine()->getManager();
-            $password = $this->get('security.password_encoder')->encodePassword($user, $user->getPassword());
-            $user->setPassword($password);
+        if ($form->isSubmitted() && $form->isValid()) {
+            // encode the plain password
+            $user->setPassword(
+                $userPasswordHasher->hashPassword(
+                    $user,
+                    $user->getPassword()
+                )
+            );
 
+            $em = $doctrine->getManager();
             $em->persist($user);
```

Comment:
tester avec et sans persist et voir les recommendation de symfony

Reply:
Fixed => See commit [Issue #4 Implementation of automated tests](https://github.com/l-lemaitre/todolist/pull/9/commits/cddb13e1ca46e6ffbaba7587e68c926c7aedbaa7)